File: kt/Minsk/src/main/kotlin/dev/phytolizer/minsk/LiteralExpressionSyntax.kt
package dev.phytolizer.minsk

class LiteralExpressionSyntax(val literalToken: SyntaxToken) : ExpressionSyntax() {
    override val kind: SyntaxKind
        get() = SyntaxKind.LiteralExpression
    override val children: List<SyntaxNode>
        get() = listOf(literalToken)
}

File: nim/src/minsk/expressions/literalExpressionSyntax.nim
import ../expressionSyntax
import ../syntaxKind
import ../syntaxNode
import ../syntaxToken

type
  LiteralExpressionSyntax* = ref object of ExpressionSyntax
    literalToken*: SyntaxToken

proc newLiteralExpressionSyntax*(literalToken: SyntaxToken): LiteralExpressionSyntax =
  new(result)
  result.literalToken = literalToken

method kind*(self: LiteralExpressionSyntax): SyntaxKind =
  return SyntaxKind.LiteralExpression

method children*(self: LiteralExpressionSyntax): seq[SyntaxNode] =
  result.add(self.literalToken)
